File: code/ex921.java
class Solution {
    public int minAddToMakeValid(String S) {
        int cnt = 0;
        Stack<Character> stack = new Stack();
        for (char c : S.toCharArray()) {
            if (c == '(')    stack.push('(');
            else {
                if (stack.isEmpty())    cnt++;
                else    stack.pop();
            }
        }
        return cnt + stack.size();
    }
}

File: code/ex921.cpp
class Solution {
public:
    int minAddToMakeValid(string S) {
        int cnt = 0;
        stack<char> stk;
        for (char c : S) {
            if (c == '(') {
                stk.push(c);
            } else {    // ')'
                if (stk.empty())    cnt++;
                else    stk.pop();
            }
        }
        return cnt + stk.size();
    }
};